src/mpi/romio/adio/common/utils.c

```diff
@@ -6,6 +6,47 @@
 #include <adio.h>
 #include <limits.h>
 
+int ADIOI_Type_get_combiner(MPI_Datatype datatype, int *combiner)
+{
+    int ret;
+#if MPI_VERSION >= 4
+    MPI_Count ni, na, nc, nt;
+    ret = MPI_Type_get_envelope_c(datatype, &ni, &na, &nc, &nt, combiner);
+#else
+    int ni, na, nt;
+    ret = MPI_Type_get_envelope(datatype, &ni, &na, &nt, combiner);
+#endif
+    return ret;
+}
+
+int ADIOI_Type_ispredef(MPI_Datatype datatype, int *flag)
```

Comment:
How about directly return the boolean value?

Reply:
Well, most of the code in ROMIO does not check error codes. That's obviously bad practice, so I did not want to add my own contribution. Additionally, you already have the query routine `ADIOI_Datatype_iscontig()` with the same signature as this one.

Reply:
Alternatively, we could implement your suggestion, and add a new one `ADIOI_Type_iscontig()` wrapping `ADIOI_Datatype_iscontig()`.

Reply:
Don't worry about it. We can always refactor later.